### ISSUE TITLE
Feature: improve dynamic importing of components

### ISSUE THREAD

**Issue by richardeschloss** [richardeschloss/nuxt-stories#63]
I think it may be useful if dynamic (lazy-loading) of components can be done from the UI. I have to brainstorm this how to make this elegant. Originally thinking of a reserved keyword "components" for the frontMatter, and if that's set, auto import those.:

```
---
components:
  - /path/to/comp1
  - /path/to/comp2
  - unpkg.com/some/remote/component3
  - chart: unpkg.com/some/remote/component4
---
```

And then the story could just use those. Components could perhaps be parsed and then use the parsed name (string after the right-most "/") or an alias if specified "alias: componentPath".

Then, those components could just be used:

<chart />

Would instantiate "unpkg.com/some/remote/component4" when ready.  Still some brainstorming to do. I'm thinking the source of components can be either: bit, npm, unpkg, github, gitlab. Maybe those shorthands should be respected if possible.

**Comment by richardeschloss** [comment]
Hmm, rethinking this a bit. May want to have a component browser that maybe floats at the side of the UI. Could be useful to know what components are available at fingertips vs remote.

**Comment by richardeschloss** [comment]
Question: is `import('https://someRemoteComponent{.js|.vue}')` discouraged? Considered an insecure design? I'm having the toughest time getting this to work in Nuxt (even though I can get it to work in other HTML pages). I tried customizing the loaders but I don't get very far. An alternative approach I've been considering is to pre-fetch the remote script, cache it and then import it with Vue.component. This approach seems to be pretty clean because then I can easily refer to the cached components by name.

You might ask: why bother? Why not just npm install stuff and be done with it? Well, my answer to that would be: sometime I have small utilities or components I want to re-use or try out in my story, but I don't want to side-step to npm install it (and wait for the app to reboot). I just want to grab the *minimal amount* I can to be able to use that remote component and just run with it asap. For example, say I have StringUtils somewhere, and I want to use it's camelCase function. I'd want to, in my story, quickly paste in that URL (pointing to StringUtils) and then when I want to use camelCase, it'd be as fast as writing StringUtils.camelCase(...)...while that app is still running.

**Comment by richardeschloss** [comment]
I have an interesting hack that may work for this. (It's working pretty fast on my end)

**Comment by richardeschloss** [comment]
Status update: I do have some cool things working on my end, but the things that work can also be thought of as dangerous. I guess an analogy would be: motorcycles can be considered dangerous too, but people still ride them super fast...

**Comment by richardeschloss** [comment]
Will be released with 2.0.14.
